src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests.cs

```diff
@@ -31413,9 +31487,9 @@ public static MyCollection<T> Create<T>(ReadOnlySpan<T> items) where T : notnull
             // We're missing diagnostics for the `where T : notnull` constraint on the Create method
             // Tracked by https://github.com/dotnet/roslyn/issues/68786
             CreateCompilation(src, targetFramework: TargetFramework.Net80).VerifyEmitDiagnostics(
-                // (7,28): warning CS8714: The type 'string?' cannot be used as type parameter 'T' in the generic type or method 'MyCollectionBuilder.Create<T>(ReadOnlySpan<T>)'. Nullability of type argument 'string?' doesn't match 'notnull' constraint.
-                // MyCollection<string?> x1 = [null]; // 1
-                Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterNotNullConstraint, "[null]").WithArguments("MyCollectionBuilder.Create<T>(System.ReadOnlySpan<T>)", "T", "string?").WithLocation(7, 28),
+                //// (7,28): warning CS8714: The type 'string?' cannot be used as type parameter 'T' in the generic type or method 'MyCollectionBuilder.Create<T>(ReadOnlySpan<T>)'. Nullability of type argument 'string?' doesn't match 'notnull' constraint.
+                //// MyCollection<string?> x1 = [null]; // 1
+                //Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterNotNullConstraint, "[null]").WithArguments("MyCollectionBuilder.Create<T>(System.ReadOnlySpan<T>)", "T", "string?").WithLocation(7, 28),
```

Comment:
>WRN_NullabilityMismatchInTypeParameterNotNullConstraint [](http://example.com/codeflow?start=39&length=55)

This case was previously handled in `Binder.GetAndValidateCollectionBuilderMethod()` in the call to `collelctionBuilderMethod.CheckConstraints()`. That was unintentional, because initial binding should check type constraints but not nullability constraints. Nullability constraints should be checked in `NullableWalker`, and the existing work item in [https://github.com/dotnet/roslyn/issues/68786](https://github.com/dotnet/roslyn/issues/68786) covers that.